File: pizza_store_api/src/main/java/fa/training/pizza_store_api/model/Product.java
package fa.training.pizza_store_api.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.List;

@Data
public class Product {
    private int id;
    private int categoryId;
    private String name;
    private String description;
    private String imageUrl;
    @JsonProperty("isAvailable")
    private boolean isAvailable;
    @JsonProperty("isDeleted")
    private boolean isDeleted;
    private LocalDateTime createAt;

    // Trường mở rộng phục vụ hiển thị
    private String categoryName;

    // Danh sách các biến thể kích thước + giá tiền của sản phẩm này
    private List<ProductVariant> variants;
}

File: pizza-store-ui/src/app/core/models/product.ts
export interface ProductVariant{
  id: number;
  productId: number;
  sizeId: number;
  price: number;
  sizeName: string;
  sizeCode: string;
}

export interface Product {
  id: number;
  categoryId: number;
  name: string;
  description: string;
  imageUrl: string;
  isAvailable: boolean;
  isDeleted: boolean;
  createdAt: string;
  variants: ProductVariant[];
}

export interface ProductVariantRequest {
  sizeId: number;
  price: number;
}

export interface ProductRequest {
  categoryId: number;
  name: string;
  description: string;
  imageUrl: string;
  isAvailable: boolean;
  variants: ProductVariantRequest[];
}
